pkg/protocol/protocol.go

```diff
@@ -289,6 +289,9 @@ func (p *Protocol) ProcessBlock(block *model.Block, src network.PeerID) error {
 			candidateEngine.ProcessBlockFromPeer(block, src)
 			if candidateEngine.IsBootstrapped() &&
 				candidateEngine.Storage.Settings().LatestCommitment().CumulativeWeight() > mainEngine.Storage.Settings().LatestCommitment().CumulativeWeight() {
+				// TODO: when activating the candidate engine should we make sure that the candidate engine produces the same commitments as the chain we decided to switch to?
```

Comment:
Seems like a reasonable sanity check, although I am not sure if it is really necessary, as we only spawn a candidate engine when we proved that a heavier chain exists, and once a heavier chain is synced (even though not the same one) you should anyway prefer this candidate engine over your main as its CW is higher!

Reply:
Addressed this in `onForkDetected`

Reply:
Addressed this in `onForkDetected`